ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -232,6 +216,90 @@ bool RSkComponentImage::processImageData(const char* path, char* response, int s
   return true;
 }
 
+inline void RSkComponentImage::drawContentShadow( SkCanvas *canvas,
+                            SkRect frameRect,
+                            SkRect targetRect,
+                            sk_sp<SkImage> imageData ,
+                            const ImageProps &imageProps){
+    if(!(layer()->componentShadow.isShadowVisible())){
+      return;
+    }
+
+    SkPaint shadowPaint;
+    filterForShadow(shadowPaint,imageProps,targetRect,frameRect,true,imageData->isOpaque());
+    bool SaveLAyerDone=false;
+    bool shadowOnFrame=(( frameRect.width() < targetRect.width()) || ( frameRect.height() < targetRect.height())||(imageProps.resizeMode == ImageResizeMode::Repeat));
+    SkMatrix identityMatrix;
+    SkRect shadowRect;
+    SkIRect imageFrameBounds;
+      //Calculate absolute frame bounds
+      if(shadowOnFrame) {
+        imageFrameBounds.setXYWH(frameRect.x() + layer()->componentShadow.shadowOffset.width(), frameRect.y() + layer()->componentShadow.shadowOffset.height(), frameRect.width(), frameRect.height());
+      } else {
+        imageFrameBounds.setXYWH(targetRect.x() + layer()->componentShadow.shadowOffset.width(), targetRect.y() + layer()->componentShadow.shadowOffset.height(), targetRect.width(), targetRect.height());
+     }
+    SkIRect shadowBounds=layer()->componentShadow.getShadowBounds(imageFrameBounds);
+    shadowRect=SkRect::Make(shadowBounds);
+
+    if(layer()->componentShadow.shadowOpacity) {
+      canvas->saveLayerAlpha(&shadowRect,layer()->componentShadow.shadowOpacity);
+      SaveLAyerDone=true;
+    }
+    if(!imageData->isOpaque() ) {
+      canvas->drawImageRect(imageData, targetRect, &shadowPaint);
+    } else {
+      if(!SaveLAyerDone) {
+        canvas->saveLayer(&shadowRect,&shadowPaint);
```

Comment:
Here also need to set "SaveLAyerDone=true" as SaveLayer been done Here. Also this blocks needs to changed as below to consider component/Image opacity

REwritting else case as below


!componentShadow.isComponentTransparent